File: 2845_Count_of_Interesting_Subarrays.java
class Solution {
    public long countInterestingSubarrays(List<Integer> nums, int modulo, int k) {
        Map<Integer, Long> freq = new HashMap<>();
        freq.put(0, 1L);  // empty prefix is valid

        long res = 0;
        int prefix = 0;

        for (int num : nums) {
            if (num % modulo == k) {
                prefix++;
            }

            int target = (prefix - k + modulo) % modulo;
            res += freq.getOrDefault(target, 0L);

            int modValue = prefix % modulo;
            freq.put(modValue, freq.getOrDefault(modValue, 0L) + 1);
        }

        return res;
    }
}

File: 2845_Count_of_Interesting_Subarrays.cpp
class Solution {
public:
    long long countInterestingSubarrays(vector<int>& nums, int modulo, int k) {
        unordered_map<int, long long> freq;
        freq[0] = 1;
        int prefixMod = 0;
        long long count = 0;

        for (int num : nums) {
            if (num % modulo == k)
                prefixMod++;

            int need = (prefixMod - k + modulo) % modulo;
            count += freq[need];

            freq[prefixMod % modulo]++;
        }

        return count;
    }
};